File: dynabuffers-python/tests/usecase/Schema20Test.py
import os
import unittest

from antlr4 import FileStream

from dynabuffers.Dynabuffers import Dynabuffers


class Schema20Test(unittest.TestCase):
    root_dir = os.path.dirname(os.path.realpath(__file__))

    def setUp(self):
        self.engine = Dynabuffers.parse(FileStream(self.root_dir + "/schema20.dbs"))

    def test_namespace(self):
        message = {"contentA":"abc"}
        result = self.engine.deserialize(self.engine.serialize(message, ["`leftshiftone/echo`","incoming"]), ["`leftshiftone/echo`","incoming"])

        self.assertEqual(message, result)


if __name__ == "__main__":
    unittest.main()

File: dynabuffers-java/src/test/kotlin/dynabuffers/usecase/Schema20Test.kt
package dynabuffers.usecase

import dynabuffers.AbstractDynabuffersTest
import dynabuffers.Dynabuffers
import dynabuffers.DynabuffersEngine
import dynabuffers.SpecialKey
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class Schema20Test : AbstractDynabuffersTest() {

    lateinit var engine: DynabuffersEngine

    @BeforeEach
    fun setUp() {
        engine = Dynabuffers.parse(Schema20Test::class.java.getResourceAsStream("/schema20.dbs"))
    }

    @Test
    fun namespaceTest() {
        val message = mapOf("contentA" to "abc")
        val serializedMessage = engine.serialize(message, listOf("`leftshiftone/echo`", "incoming"))

        val result = engine.deserialize(serializedMessage, listOf("`leftshiftone/echo`", "incoming"))

        assertMap(message.plus(SpecialKey.NAMESPACE.key to "`leftshiftone/echo`.incoming"), result)
    }

}
